packages/ia-components/sandbox/audio-players/audio-player-main.jsx

```diff
@@ -72,16 +73,31 @@ export default class TheatreAudioPlayer extends Component {
    */
   showMedia() {
     const { source, sourceData } = this.props;
-    const isExternal = source === 'youtube' || source === 'spotify';
     let mediaElement = (
       <ArchiveAudioPlayer
         {...this.props}
         onRegistrationComplete={this.receiveURLSetter}
         needsURLSettingAccess
       />
-    );
-    if (isExternal) {
-      // make iframe with URL
+    ); // shorten code
+    if (source === 'youtube') {
+      const { urlSetterFN } = this.state;
```

Comment:
This is being extracted in both conditions and passed a `trackNumber` if it's defined. This definition, the definition of `trackNumber`, and the resulting conditional to call `urlSetterFN` can be pulled out of both of these parent conditionals and simply performed after both of them.

Reply:
I'll be using `trackNumber` in the youtube player so I'll take the rest of the definitions and conditional out.

Reply:
Additionally `urlSetterFN` also returns a value for archive and executes (and returns an error) for it when it's outside the youtube and spotify conditionals